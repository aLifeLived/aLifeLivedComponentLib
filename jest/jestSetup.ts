/* eslint-disable no-undef */
jest.useFakeTimers({ legacyFakeTimers: true }); // Stops reference error from appearing
