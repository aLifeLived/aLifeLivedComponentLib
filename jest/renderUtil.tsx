import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';
import { library } from '@fortawesome/fontawesome-svg-core';
import { iconList } from '../src/themes/systemIcon/IconList';

library.add(iconList);

function render(ui: any, { ...renderOptions } = {}) {
  function Wrapper({ children }: { children: any }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
