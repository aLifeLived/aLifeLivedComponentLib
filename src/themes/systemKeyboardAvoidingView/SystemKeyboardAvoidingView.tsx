import {
  backgroundColor,
  BackgroundColorProps,
  border,
  BorderProps,
  color,
  ColorProps,
  createRestyleComponent,
  layout,
  LayoutProps,
  spacing,
  SpacingProps,
} from '@shopify/restyle';
import React from 'react';
import { KeyboardAvoidingView, KeyboardAvoidingViewProps } from 'react-native';
import { Theme } from '../theme';

export type StyledKeyboardAvoidingViewProps = KeyboardAvoidingViewProps &
  SpacingProps<Theme> &
  LayoutProps<Theme> &
  BorderProps<Theme> &
  BackgroundColorProps<Theme> &
  ColorProps<Theme> & { children?: React.ReactNode };

export const SystemKeyboardAvoidingView = createRestyleComponent<
  StyledKeyboardAvoidingViewProps,
  Theme
>([spacing, layout, border, backgroundColor, color], KeyboardAvoidingView);
