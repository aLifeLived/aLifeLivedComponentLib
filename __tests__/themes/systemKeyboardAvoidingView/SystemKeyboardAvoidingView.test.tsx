import React from 'react';
import { render } from '../../../jest/renderUtil';

// Components
import { SystemKeyboardAvoidingView } from '../../../src/themes/systemKeyboardAvoidingView/SystemKeyboardAvoidingView';

const setup = () => render(<SystemKeyboardAvoidingView flex={1} />);

describe('themes >> systemKeyboardAvoidingView', () => {
  describe('Given style has been applied to avoiding view', () => {
    test('Should match snapshot with flex of 1', () => {
      const { toJSON } = setup();
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
