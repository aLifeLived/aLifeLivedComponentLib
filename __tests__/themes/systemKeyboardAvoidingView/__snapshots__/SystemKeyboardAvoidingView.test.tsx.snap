// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themes >> systemKeyboardAvoidingView Given style has been applied to avoiding view Should match snapshot with flex of 1 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
    ]
  }
/>
`;
